File: src/test/scala/com/carmanconsulting/akka/scala/HelloAkkaTest.scala
package com.carmanconsulting.akka.scala

import akka.actor.Props
import org.junit.Test

class HelloAkkaTest extends AkkaTestCase {

  @Test
  def testHelloAkka(): Unit = {
    val hello = system.actorOf(Props[HelloAkka])
    hello ! "Akka"
    expectMsg("Hello, Akka!")
  }
}

File: src/test/java/com/carmanconsulting/akka/java/HelloAkkaTest.java
package com.carmanconsulting.akka.java;

import akka.actor.ActorRef;
import akka.actor.Props;
import org.junit.Test;

public class HelloAkkaTest extends AkkaTestCase {

    @Test
    public void testHelloAkka() {
        ActorRef hello = system().actorOf(Props.create(HelloAkka.class));
        hello.tell("Akka", testActor());
        expectMsg("Hello, Akka!");
    }
}
